### PR TITLE
[Cherry pick #2402 -> 1.26] Add some validations for L4 multinetwork services.

### PR DESCRIPTION
**Pull request by mmamczur**
Cherry pick of #2402

- only services in network type L3 are supported.
- multinet services with externalTrafficPolicy other than Local are not supported for now.

### CODE REVIEW

**Review by swetharepakula — COMMENTED** [review]
/hold cancel

/lgtm

File: pkg/network/network.go [hunk 1]
```diff
@@ -100,7 +107,7 @@ func (nr *NetworksResolver) ServiceNetwork(service *apiv1.Service) (*NetworkInfo
 		return nil, utils.NewUserError(fmt.Errorf("GKENetworkParamSet %s was not found", parametersRef.Name))
 	}
 	gkeNetworkParamSet := gkeParamsObj.(*networkv1.GKENetworkParamSet)
-	if network == nil {
+	if gkeNetworkParamSet == nil {
```
[the remark this answers is not in the document]
Comment (mmamczur) [reply pkg/network/network.go:110]: we don't, and actually I stared checking out how we handle cases like this in our code

usually we don't check if the cast worked
examples (most common):
https://github.com/kubernetes/ingress-gce/blob/master/pkg/psc/controller.go#L290C10-L290C10

https://github.com/kubernetes/ingress-gce/blob/master/pkg/neg/controller.go#L758

sometimes we do like this
https://github.com/kubernetes/ingress-gce/blob/master/pkg/neg/controller.go#L469C10-L469C10

and sometimes with a bool
https://github.com/kubernetes/ingress-gce/blob/master/pkg/neg/controller.go#L710C10-L710C10

the interesting bit is that this variant
```
service := obj.(*apiv1.Service)
```
 can result in a panic
 
 only this seems to work well:
 ```
 service, ok := obj.(*apiv1.Service)
 ```
 
 the question is: should we check if most code doesn't? I guess we'd have many issues already if these conversions ever failed

**Pull request merged**